testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/security/authentication/BasicAuthMechanismServerSetupTask.java

```diff
@@ -0,0 +1,152 @@
+        operations.add(undefineAttrOp2);
+
+        ModelNode addOp3 = createOpNode("core-service=management/access=identity", ADD);
+        addOp3.get("security-domain").set("ManagementDomain");
+        operations.add(addOp3);
+        ModelNode writeAttrOp4 = createOpNode("core-service=management/management-interface=http-interface", WRITE_ATTRIBUTE_OPERATION);
+        writeAttrOp4.get(ClientConstants.NAME).set("http-upgrade");
+        writeAttrOp4.get(ClientConstants.VALUE).add("enabled", true);
+        writeAttrOp4.get(ClientConstants.VALUE).add("sasl-authentication-factory", "management-sasl-authentication");
+        operations.add(writeAttrOp4);
+        ModelNode writeAttrOp5 = createOpNode("core-service=management/management-interface=http-interface", WRITE_ATTRIBUTE_OPERATION);
+        writeAttrOp5.get(ClientConstants.NAME).set("http-authentication-factory");
+        writeAttrOp5.get(ClientConstants.VALUE).set("management-http-authentication");
+        operations.add(writeAttrOp5);
+        ModelNode undefineAttrOp3 = createOpNode("core-service=management/management-interface=http-interface", UNDEFINE_ATTRIBUTE_OPERATION);
+        undefineAttrOp3.get(ClientConstants.NAME).set("security-realm");
+        operations.add(undefineAttrOp3);
+        ModelNode removeOp1 = createOpNode("core-service=management/security-realm=ApplicationRealm/authentication=local", REMOVE);
+        operations.add(removeOp1);
+        ModelNode removeOp2 = createOpNode("core-service=management/security-realm=ApplicationRealm/authentication=properties", REMOVE);
+        operations.add(removeOp2);
+        ModelNode removeOp3 = createOpNode("core-service=management/security-realm=ApplicationRealm/authorization=properties", REMOVE);
+        operations.add(removeOp3);
+
+        ModelNode updateOp = Operations.createCompositeOperation(operations);
+        updateOp.get(OPERATION_HEADERS, ROLLBACK_ON_RUNTIME_FAILURE).set(false);
+        updateOp.get(OPERATION_HEADERS, ALLOW_RESOURCE_SERVICE_RESTART).set(true);
+        CoreUtils.applyUpdate(updateOp, managementClient.getControllerClient());
+
+        List<ModelNode> operations1 = new ArrayList<>();
+        ModelNode removeOp4 = createOpNode("subsystem=security", REMOVE);
+        operations1.add(removeOp4);
+
+        ModelNode updateOp1 = Operations.createCompositeOperation(operations1);
+        updateOp.get(OPERATION_HEADERS, ROLLBACK_ON_RUNTIME_FAILURE).set(false);
+        updateOp.get(OPERATION_HEADERS, ALLOW_RESOURCE_SERVICE_RESTART).set(true);
+        CoreUtils.applyUpdate(updateOp1, managementClient.getControllerClient());
+    }
+
+    /**
+     * Recreates the security subsystem
+     *
+     * @param managementClient
+     * @param containerId
+     * @throws Exception
+     */
+    @Override
+    public void tearDown(ManagementClient managementClient, String containerId) throws Exception {
+        List<ModelNode> operations = new ArrayList<>();
```

Comment:
In general this is way to fragile, it would be very easy for this test to not restore the security subsystem to the default config, especially if the config every changes. With that in mind I have come up with a new way to handle restoring server state when a reload is required by using snapshots: https://github.com/wildfly/wildfly/pull/11260

Once that PR is merged this should be modified to use SnapshotRestoreSetupTask to make sure all changes are correctly reverted